codegenerator/cli/npm/envio/src/PgStorage.res

```diff
@@ -143,11 +143,238 @@ END IF;`
     // Return optimized queries - main DDL in DO block, functions separate
     // Note: DO $$ BEGIN wrapper is only needed for PL/pgSQL conditionals (IF NOT EXISTS)
     // Reset case uses direct DDL (faster), non-cleanRun case uses conditionals (safer)
-    cleanRun ? query.contents : `DO $$ BEGIN ${query.contents} END $$;`,
+    cleanRun || enums->Utils.Array.isEmpty
+      ? query.contents
+      : `DO $$ BEGIN ${query.contents} END $$;`,
     // Functions query (separate as they can't be in DO block)
   ]->Js.Array2.concat(functionsQuery.contents !== "" ? [functionsQuery.contents] : [])
 }
 
+let makeLoadByIdSql = (~pgSchema, ~tableName) => {
+  `SELECT * FROM "${pgSchema}"."${tableName}" WHERE id = $1 LIMIT 1;`
+}
+
+let makeLoadByIdsSql = (~pgSchema, ~tableName) => {
+  `SELECT * FROM "${pgSchema}"."${tableName}" WHERE id = ANY($1::text[]);`
+}
+
+let makeInsertUnnestSetSql = (~pgSchema, ~table: Table.table, ~itemSchema, ~isRawEvents) => {
+  let {quotedFieldNames, quotedNonPrimaryFieldNames, arrayFieldTypes} =
+    table->Table.toSqlParams(~schema=itemSchema)
+
+  let primaryKeyFieldNames = Table.getPrimaryKeyFieldNames(table)
+
+  `INSERT INTO "${pgSchema}"."${table.tableName}" (${quotedFieldNames->Js.Array2.joinWith(", ")})
+SELECT * FROM unnest(${arrayFieldTypes
+    ->Js.Array2.mapi((arrayFieldType, idx) => {
+      `$${(idx + 1)->Js.Int.toString}::${arrayFieldType}`
+    })
+    ->Js.Array2.joinWith(",")})` ++
+  switch (isRawEvents, primaryKeyFieldNames) {
+  | (true, _)
+  | (_, []) => ``
+  | (false, primaryKeyFieldNames) =>
+    `ON CONFLICT(${primaryKeyFieldNames
+      ->Js.Array2.map(fieldName => `"${fieldName}"`)
+      ->Js.Array2.joinWith(",")}) DO ` ++ (
+      quotedNonPrimaryFieldNames->Utils.Array.isEmpty
+        ? `NOTHING`
+        : `UPDATE SET ${quotedNonPrimaryFieldNames
+            ->Js.Array2.map(fieldName => {
+              `${fieldName} = EXCLUDED.${fieldName}`
+            })
+            ->Js.Array2.joinWith(",")}`
+    )
+  } ++ ";"
+}
+
+let makeInsertValuesSetSql = (~pgSchema, ~table: Table.table, ~itemSchema, ~itemsCount) => {
+  let {quotedFieldNames, quotedNonPrimaryFieldNames} = table->Table.toSqlParams(~schema=itemSchema)
+
+  let primaryKeyFieldNames = Table.getPrimaryKeyFieldNames(table)
+  let fieldsCount = quotedFieldNames->Array.length
+
+  // Create placeholder variables for the VALUES clause - using $1, $2, etc.
+  let placeholders = ref("")
+  for idx in 1 to itemsCount {
+    if idx > 1 {
+      placeholders := placeholders.contents ++ ","
+    }
+    placeholders := placeholders.contents ++ "("
+    for fieldIdx in 0 to fieldsCount - 1 {
+      if fieldIdx > 0 {
+        placeholders := placeholders.contents ++ ","
+      }
+      placeholders := placeholders.contents ++ `$${(fieldIdx * itemsCount + idx)->Js.Int.toString}`
+    }
+    placeholders := placeholders.contents ++ ")"
+  }
```

Comment:
Finding this logic quite tough to review. What about simplifying like this?
```suggestion
  // Create placeholder variables for the VALUES clause - using $1, $2, etc.
let items = []
for idx in 1 to itemsCount * fieldsCount {
  let _ = items->Js.Array2.push(`$${idx->Belt.Int.toString}`)
}
let placeholders = `(${items->Js.Array2.joinWith(",")})`
```

Reply:
No, this is a completely different result.